src/panel_gwalker/_gwalker.py

```diff
@@ -54,28 +222,112 @@ class GraphicWalker(ReactComponent):
+        objects=["explorer", "profiler", "viewer", "chart"],
+        doc="""How to display the data. One of 'explorer' (default), 'profiler,
+        'viewer' or 'chart'.""",
+    )
+    index: int | list[int] | None = param.ClassSelector(
+        class_=(int, list, type(None)),
+        doc="""An optional chart index or list of chart indices to display in the 'viewer' or 'chart' renderer.
+    Has no effect on other renderers.""",
+    )
+    page_size: int = param.Integer(
+        20,
+        bounds=(1, None),
+        doc="""The number of rows per page in the table of the 'profiler' render.
+    Has no effect on other renderers.""",
+    )
+    tab: Literal["data", "vis"] = param.Selector(
+        default="vis",
+        objects=["data", "vis"],
+        doc="""Set the active tab to 'data' or 'vis' (default). Only applicable for the 'explorer' renderer. Not bi-directionally synced with client.""",
+    )
+    container_height: str = param.String(
+        default="400px",
+        doc="""The height of a single chart in the 'viewer' or 'chart' renderer. For example '500px' (pixels) or '30vh' (viewport height).""",
+    )
 
     appearance: Literal["media", "dark", "light"] = param.Selector(
         default="light",
         objects=["light", "dark", "media"],
         doc="""Dark mode preference: 'light', 'dark' or 'media'.
         If not provided the appearance is derived from pn.config.theme.""",
     )
-    theme: Literal["vega", "g2", "streamlit"]=param.Selector(
-        default="vega",
-        objects=["vega", "g2", "streamlit"],
-        doc="""The theme of the chart(s). One of 'vega' (default), 'g2' or 'streamlit'.""",
+    theme_key: Literal["g2", "streamlit", "vega"] = param.Selector(
+        default="g2",
+        objects=["g2", "streamlit", "vega"],
+        doc="""The theme of the chart(s). One of 'g2', 'streamlit' or 'vega' (default).""",
+    )
+
+    export_mode: Literal["spec", "svg"] = param.Selector(
+        label="Mode",
+        default="spec",
+        objects=["spec", "svg"],
+        doc="""Used as default mode for export and save methods.""",
+    )
+    export_scope: Literal["all", "current"] = param.Selector(
+        label="Scope",
+        default="all",
+        objects=["all", "current"],
+        doc="""Used as default scope for export and save methods.""",
+    )
+    export_timeout: int = param.Integer(
+        label="Timeout",
+        default=5000,
+        doc="""Export timeout in milliseconds. Used as default for export and save methods.""",
+    )
+    export: Callable[
+        Concatenate[
+            Literal["spec", "svg", None],
+            Literal["current", "all", None],
+            Optional[int],
+            P,
+        ],
+        Coroutine[Any, Any, str],
+    ] = param.Action(
+        doc="""Exports the chart(s) as either a spec or SVG.""",
+        constant=True,
+        allow_refs=False,
+    )
+
+    save_path: str | PathLike = param.ClassSelector(
```

Comment:
Same with these.

Reply:
Same reply :-) This is also used as a default argument for the `save` method. Not only for the control.

Reply:
This gives flexibility and speed.

In this way you can easily define what and where you want to save in the `__init__` method and then put in a single button if you don't want to provide options to users. Or you can easily provide the options to users.

Reply:
Try to make a non-basic application where you want to sprinkle in options to configure export and save and enable users to export and save. That takes time if you have to do it from scratch. That is what I'm trying to avoid.